src/utils/storage.js

```diff
@@ -145,12 +145,26 @@ export function writeMetadataToDb({
       })
 
       // Function for creating promise for writing file metadata (handles writing to RTDB or Firestore)
+      const documentIdFromOptions =
+        typeof documentId === 'function'
+          ? documentId(
+              uploadTaskSnapshot,
+              firebase,
+              uploadTaskSnapshot.metadata,
+              downloadURL
+            )
+          : documentId
       const metaSetPromise = (fileData) => {
         if (useFirestoreForStorageMeta) {
-          return firebase // Write metadata to Firestore
-            .firestore()
-            .collection(dbPath)
-            .add(fileData)
+          if (documentIdFromOptions) {
+            const docRef = firebase // Write metadata to Firestore
+              .firestore()
+              .collection(dbPath)
+              .doc(documentIdFromOptions)
+            return docRef.update(fileData).then(() => docRef)
+          } else {
```

Comment:
We shouldn't need the else here since we are returning in the if

Reply:
Fair enough, sometimes I think it's more readable with the `else`, but you 're right that it's not necessary. I will make this change.